gammapy/modeling/models/cube.py

```diff
@@ -767,8 +768,9 @@ def evaluate(self):
             Background evaluated on the Map
         """
         value = self.spectral_model(self.energy_center).value
-        back_values = self.map.data * value
-        return self.map.copy(data=back_values)
+        self._evaluation_cache.data = self.map.data * value
```

Comment:
I'd say we don't need the caching on the `.evaluate()` level, this is only used for independent evaluation and not used during fitting.

Reply:
for the `TemplateNPredModel` the evaluator call `.evaluate()` 
https://github.com/gammapy/gammapy/blob/19da7840b81b88092d42cd5ea032a7f9a9b7276f/gammapy/datasets/evaluator.py#L361

Reply:
Yes, but we have caching on the `MapEvaluator`, so the effect on the performance should be completely negligible.

Reply:
True

Reply:
removed